x/cdp/abci.go

```diff
@@ -21,6 +21,10 @@ func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k Keeper) {
 			k.HandleNewDebt(ctx, cp.Denom, dp.Denom, timeElapsed)
 		}
 
+		// IMPLEMENT HERE FEE UPDATING
+		// call our update fees method for the risky cdps
+		k.UpdateFeesForRiskyCdps(ctx, cp)
```

Comment:
@karzak Is just calling the function here OK? Since it updates the state of the CDPs I assuming just calling is sufficient and no need to do any other state updates?

Reply:
Yes, but I would have the function return an error and follow the pattern of emitting the error if it's not `nil` (see how errors are emitted a few lines below)

```
if err != nil {
			ctx.EventManager().EmitEvent(
				sdk.NewEvent(
					EventTypeBeginBlockerFatal,
					sdk.NewAttribute(sdk.AttributeKeyModule, fmt.Sprintf("%s", ModuleName)),
					sdk.NewAttribute(types.AttributeKeyError, fmt.Sprintf("%s", err)),
				),
			)
		}
```